components-java-sdk/src/main/java/io/dapr/components/domain/state/GetResponse.java

```diff
@@ -25,21 +29,50 @@
  * @param metadata Metadata related to the response.
  * @param contentType The response value contenttype
  */
-public record GetResponse(byte[] data, String etag, Map<String, String> metadata, String contentType) {
+public record GetResponse(ByteString data, String etag, Map<String, String> metadata, String contentType) {
 
   /**
-   * Constructor.
+   * Canonical Constructor.
    *
    * @param data The value of the GetRequest response.
    * @param etag The etag of the associated key.
    * @param metadata Metadata related to the response.
    * @param contentType The response value contenttype
    */
-  public GetResponse(byte[] data, String etag, Map<String, String> metadata, String contentType) {
+  public GetResponse(final ByteString data, final String etag, final Map<String, String> metadata,
+                     final String contentType) {
+    // Java lacks a solution to "immutable byte arrays", so we wrap them up in a ByteString.
+    // FindBugs isn't really happy with this setup so we exclude EI_EXPOSE_REP* errors for this class
     this.data = Objects.requireNonNull(data);
     this.etag = Objects.requireNonNull(etag);
     // All this constructor just so we can make this Map unmodifiable and this class immutable ;)
     this.metadata = Collections.unmodifiableMap(Objects.requireNonNull(metadata));
     this.contentType = Objects.requireNonNull(contentType);
   }
+
+  /**
+   * Constructor.
+   *
+   * @param data The value of the GetRequest response.
+   * @param etag The etag of the associated key.
+   * @param metadata Metadata related to the response.
+   * @param contentType The response value contenttype
+   */
+  public GetResponse(final byte[] data, final String etag, final Map<String, String> metadata,
```

Comment:
I like the override that still uses the `byte[]` array.  Let developers choose the most convenient.

Reply:
I have the feeling that _I_ would ended up using that version of the API, thought it was better to leave it.